drivers/adc/adc_nrfx_saadc.c

```diff
@@ -212,6 +217,11 @@ static int input_assign(nrf_saadc_input_t *pin_p,
 		*pin_n = channel_cfg->input_negative == NRF_SAADC_GND ?
 			 NRF_SAADC_INPUT_DISABLED :
 			 saadc_psels[channel_cfg->input_negative];
+
+#if NRF_GPIO_HAS_RETENTION_SETCLEAR
+		nrf_gpio_pin_retain_disable(saadc_psels[channel_cfg->input_negative]);
```

Comment:
This function should not be called when `channel_cfg->input_negative == NRF_SAADC_GND`.